sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -331,8 +331,8 @@ class JoinSuite extends QueryTest with SharedSQLContext {
   }
 
   test("full outer join") {
-    upperCaseData.where('N <= 4).registerTempTable("left")
-    upperCaseData.where('N >= 3).registerTempTable("right")
+    upperCaseData.where('N <= 4).registerTempTable("`left`")
```

Comment:
`LEFT`/`RIGHT` are SQL keywords. The old parser is a bit more lenient.

The question is if we should allow SQL keywords in table names? I'd prefer not support this. If we do, then we should move identifier parsing into a seperate parser.